atlas-db-controller/databaseResource.go

```diff
@@ -144,16 +144,18 @@ func (c *Controller) syncDatabase(key string) error {
 		}
 	}
 
-	err = p.SyncDatabase(db, dsn)
+	state, err := p.SyncDatabase(db, dsn)
 	if err != nil {
 		msg := fmt.Sprintf("error syncing database '%s': %s", key, err)
 		glog.Error(msg)
 		c.updateDatabaseStatus(key, db, StateError, msg)
 		runtime.HandleError(fmt.Errorf(msg))
 		return err
 	}
-	msg := fmt.Sprintf("Database '%s' & users synced successfully", s.Name)
-	c.recorder.Event(db, corev1.EventTypeNormal, SuccessSynced, msg)
+	if state == StateCreated {
```

Comment:
If error is nil , does it not mean the database is created ? looks like state is always going to be `StateCreated` if `err` is nil